### ISSUE TITLE
Unable to write/read files in nested subfolders under auto_gpt_workspace on WIndows

### ISSUE THREAD

**Issue by nitefliteuk** [Significant-Gravitas/Auto-GPT#2257]
### ⚠️ Search for existing issues first ⚠️

- [X] I have searched the existing issues, and there is no existing issue for my problem

### GPT-3 or GPT-4

- [ ] I am using Auto-GPT with GPT-3 (GPT-3.5)

### Steps to reproduce 🕹

OS: Windows 10 Pro 22H2
Ask to create any python output files inside (nested) subfolders underauto_gpt_workspace. Create the subfolder if necessary. This has never worked.
Even after creating the subfolder manually and asked to save and then execute a .py file in the subfolder, it fails.

### Current behavior 😯

Operations in subfolders seem to fail possibly because of the wrong path separator. When I pointed out that this was a WIndows computer in a human feedback,  the next operation worked. The feedback was forgotten soon after though.

### Expected behavior 🤔

Should be able to create (nested) subfolders e.g. 'projectname\data\data.csv' and read and write any file inside any of them

### Your prompt 📝

```yaml
# Paste your prompt here
```

**Comment by k-boikov** [comment]
Can you please provide the prompt and the actions?

**Comment by k-boikov** [comment]
Maybe related https://github.com/Significant-Gravitas/Auto-GPT/issues/2006 https://github.com/Significant-Gravitas/Auto-GPT/issues/1966

**Comment by epicshardz** [comment]
I second this issue

**Comment by k-boikov** [comment]
Actually.... I think we are missing a "not" in front of `if we_are_running_in_a_docker_container():`

**Comment by k-boikov** [comment]
Fixed with https://github.com/Significant-Gravitas/Auto-GPT/pull/2262

**Comment by Pwuts** [comment]
@nitefliteuk please add your prompt+log and reopen the issue if it is still relevant.

**Comment by nitefliteuk** [comment]
@Pwuts Using stable version v0.2.1, here prompt and log showing the problem. Hopefully, that is detailed enough to reproduce, but let me know if there's any additional info or test I can provide.

================================== Prompt =================================
ai_goals:
- Fetch any relevant dataset from https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/
- Import the csv datasets and use the last column as output
- Develop the most accurate machine learning model using TensorFlow and Keras to maximise the
  output accuracy
- Write a Jupyter notebook including all code to import datasets set up the best model
  found in previous step and presenting results both numerically and graphically.
  The file should be saved under the folder 'wine-quality\output'
ai_name: Wine tester
ai_role: An AI to autonomously improve wine detection results from a well-known dataset.
  Any python file required to achieve the goal should be saved in the folder 'wine-quality'
  and any dataset stored locally under the folder 'wine-quality\data'

==================================== Log =================================
2023-04-18 12:47:24,561 INFO Welcome to Auto-GPT!  Enter the name of your AI and its role below. Entering nothing will load defaults.
2023-04-18 12:47:25,060 INFO Name your AI:  For example, 'Entrepreneur-GPT'
2023-04-18 12:49:52,750 INFO Wine tester here! I am at your service.
2023-04-18 12:49:53,020 INFO Describe your AI's role:  For example, 'an AI designed to autonomously develop and run businesses with the sole goal of increasing your net worth.'
2023-04-18 12:53:07,039 INFO Enter up to 5 goals for your AI:  For example: 
Increase net worth, Grow Twitter Account, Develop and manage multiple businesses autonomously'
2023-04-18 13:18:08,740 INFO Using memory of type: LocalCache
2023-04-18 13:18:08,933 INFO Using Browser: chrome
2023-04-18 13:18:21,999 INFO  THOUGHTS: Let's start by fetching the dataset from the url https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/
2023-04-18 13:18:22,348 INFO REASONING: We need to fetch the dataset before we can import and use it.
2023-04-18 13:18:22,733 INFO PLAN: 
2023-04-18 13:18:22,825 INFO -  Use the 'browse_website' command to access the website.
2023-04-18 13:18:23,147 INFO -  Use the 'search_files' command to confirm successful download of the dataset.
2023-04-18 13:18:23,560 INFO CRITICISM: None
2023-04-18 13:18:23,671 INFO SPEAK: We will browse to the dataset url and check for successful download of the dataset.
2023-04-18 13:18:24,188 INFO Attempting to fix JSON by finding outermost brackets
 
2023-04-18 13:18:24,493 INFO Apparently json was fixed. 
2023-04-18 13:18:24,673 INFO NEXT ACTION:  COMMAND = browse_website  ARGUMENTS = {'url': 'https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/', 'question': 'How to download the winequality dataset in csv format from this website?'}
2023-04-18 13:18:33,602 INFO -=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= 
2023-04-18 13:18:49,142 INFO SYSTEM:  Command browse_website returned: ("Answer gathered from website: The winequality dataset in csv format can be downloaded from the website using the provided links for winequality-red.csv and winequality-white.csv. \n \n Links: [' Parent Directory (https://archive.ics.uci.edu/ml/machine-learning-databases/)', ' winequality-red.csv (https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv)', ' winequality-white.csv (https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv)', ' winequality.names (https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality.names)']", <selenium.webdriver.chrome.webdriver.WebDriver (session="a185ad7e4d3c53afb388c37163333806")>)
2023-04-18 13:19:04,722 INFO  THOUGHTS: Let's download the dataset and confirm that the dataset is successfully downloaded
2023-04-18 13:19:05,137 INFO REASONING: The next step is to download the dataset that we found on the website, after that we can confirm the download by checking if the csv files are in the directory.
2023-04-18 13:19:05,885 INFO PLAN: 
2023-04-18 13:19:05,915 INFO -  Use the 'execute_shell' command to download the dataset by using the 'wget' command.
2023-04-18 13:19:06,392 INFO -  Use the 'search_files' command to confirm successful download of the dataset.
2023-04-18 13:19:06,748 INFO CRITICISM: None
2023-04-18 13:19:06,855 INFO SPEAK: We will download and check if we have the dataset locally.
2023-04-18 13:19:07,254 INFO Attempting to fix JSON by finding outermost brackets
 
2023-04-18 13:19:07,621 INFO Apparently json was fixed. 
2023-04-18 13:19:07,789 INFO NEXT ACTION:  COMMAND = execute_shell  ARGUMENTS = {'command_line': 'mkdir -p wine-quality/data && wget -P wine-quality/data/ https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv && wget -P wine-quality/data/ https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv'}
2023-04-18 13:19:08,639 INFO SYSTEM:  Command execute_shell returned: STDOUT:
b''
STDERR:
b'The syntax of the command is incorrect.\r\n'
